### PR TITLE
Feat/py secrets manager store

### PR DESCRIPTION
**Pull request by jamesL92**
Implementation of AWS secrets manager store for Python. Includes a base key-value store class, which can in future be used for other stores (e.g. Azure Key Valult, GCP Secrets Manager, Hashicorp Vault and so on). 

Implementation is heavily based on the Node implementation.

### CODE REVIEW

**Comment by jamesL92** [comment]
Made a few additional changes:
- unfortunately can't store `boto3.client` as an instance member on `AWSSecretsManagerStore`, since `boto3.client` has a `threading.Lock` member which can't be serialised, i.e. it can't be passed through to a thread as part of `EvalCommand._eval_from_parameters`.
- Small change in where `ConfigStore` is imported from in `key_value_store.py` to make it  consistent with the rest of the stores (was causing a weird issue whereby it wasn't being considered an instance of `ConfigStore` by `pydantic`, meaning it couldn't be used in `EvalCommandParameters`.

**Comment by RonConfigu** [comment]
Hi JamesL92,

I wanted to commend you on the outstanding work you’ve done thus far. Your contributions are greatly appreciated.

We have recently made some modifications to the SDK (@configu/ts), and I am currently working on updating the py-sdk accordingly. It seems highly likely that the threading functionality will be removed, which means your previous implementation should align perfectly without requiring you to rebuild the client for each call. (see https://github.com/configu/configu/issues/129)

I anticipate that the new version will be published no later than tomorrow, and once that happens, we will proceed to make your pull request operational and ready for merging.

Thank you once again for your efforts.

Best regards,
Ron

File: py/configu/stores/aws_secrets_manager_store.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    def get_by_key(self, key: str) -> str:
+        return self.client.get_secret_value(SecretId=key).get(
+            "SecretString", ""
+        )
+
+    def upsert(self, key: str, value: str):
+        try:
+            self.client.update_secret(SecretId=key, SecretString=value)
+        except Exception:
```
Comment (RonConfigu) [inline py/configu/stores/aws_secrets_manager_store.py:23]:
```suggestion
        except (Exception,):
```
NIT

**Review by RonConfigu — CHANGES_REQUESTED** [review]
🚀 💯 Great job! 

The added docstrings are amazing. Thank you for that! much needed.
Please review my comments and apply the suggestions. once done. we're good to go!

Thanks,
Ron

File: py/configu/stores/key_value_store.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        """
+        try:
+            return json.loads(value)
+        except Exception:
+            return {}
+
+    def calc_key(self, query: Union[ConfigStoreQuery, Config]) -> str:
+        """
+        Calculate the storage system key referenced in a query.
+        If `set` is populated, will use that. Otherwise defaults to `key`
```
[the remark this answers is not in the document]
Comment (jamesL92) [reply py/configu/stores/key_value_store.py:55]: :+1: makes perfect sense, thanks for the explanation :)

**Comment by jamesL92** [comment]
@RonConfigu I've made the requested changes, please let me know if any further changes are required :+1:

**Review by RonConfigu — APPROVED** [review]
Beautiful work. 🥇 
Thank you for your contribution @jamesL92!

We would be delighted to mention your contribution and show you our appreciation on our LinkedIn. If you are interested, please provide us with your LinkedIn account details or email.

**Pull request merged**